rust_src/src/obarray.rs

```diff
@@ -13,22 +15,13 @@ use lisp::defsubr;
 pub struct LispObarrayRef(LispObject);
 
 impl LispObarrayRef {
-    /// Return a reference to the Lisp variable `obarray`.
-    pub fn constant_obarray() -> LispObarrayRef {
-        LispObarrayRef(LispObject::from_raw(unsafe { check_vobarray() }))
+    pub fn new(obj: LispObject) -> LispObarrayRef {
+        LispObarrayRef(obj)
     }
 
-    /// Return a reference corresponding to OBJECT, or raise an error if it is
-    /// not a `LispObarrayRef`.
-    ///
-    /// # C Porting Notes
-    ///
-    /// This is a wrapper around the C function `check_obarray` instead of a
-    /// translation, so that fatal errors are handled in the same fashion.
-    pub fn from_object_or_error(object: LispObject) -> LispObarrayRef {
-        LispObarrayRef(LispObject::from_raw(unsafe {
-            check_obarray(object.to_raw())
-        }))
+    /// Return a reference to the Lisp variable `obarray`.
+    pub fn constant_obarray() -> LispObarrayRef {
+        LispObarrayRef(check_vobarray())
```

Comment:
This is a great factorization, but the name confused me. Since this is checking a global variable rather than producing a constant value, perhaps `global_obarray` would be better?

Reply:
I did not name it :-) I had left it alone for now. Let me give it another look.